convex/dashboard.ts

```
@@ -0,0 +1,445 @@
+
+export const listEvents = query({
+  args: {
+    paginationOpts: paginationOptsValidator,
+    search: v.optional(v.string()),
+    type: v.optional(
+      v.union(v.literal("pageview"), v.literal("interaction"), v.literal("error"), v.literal("custom"))
+    ),
+  },
+  handler: async (ctx, args) => {
+    if (args.search && args.search.trim().length > 0) {
+      let sq = ctx.db.query("events").withSearchIndex("search_name", (q) => {
```

ESLint

[error] 'sq' is never reassigned. Use 'const' instead.

(prefer-const)